lib/matplotlib/tests/test_axes.py

```diff
@@ -8898,3 +8898,41 @@ def test_axhvlinespan_interpolation():
     ax.axhline(1, c="C0", alpha=.5)
     ax.axhspan(.8, .9, fc="C1", alpha=.5)
     ax.axhspan(.6, .7, .8, .9, fc="C2", alpha=.5)
+
+
+@check_figures_equal(extensions=["png"])
+@pytest.mark.parametrize("which", ("x", "y"))
+def test_axes_clear_behavior(fig_ref, fig_test, which):
+    ax_test = fig_test.subplots()
+    ax_ref = fig_ref.subplots()
+    target = {
+        "direction": "in",
+        "length": 10,
+        "width": 10,
+        "color": "xkcd:wine red",
+        "pad": 0,
+        "labelfontfamily": "serif",
+        "zorder": 7,
+        "labelrotation": 45,
+        "labelcolor": "xkcd:shocking pink",
+        # this overrides color + labelcolor, skip
+        # colors: ,
+        "grid_color": "xkcd:fluorescent green",
+        "grid_alpha": 0.5,
+        "grid_linewidth": 3,
+        "grid_linestyle": ":",
+        "bottom": False,
+        "top": True,
+        "left": False,
+        "right": True,
+        "labelbottom": True,
+        "labeltop": True,
+        "labelleft": True,
+        "labelright": True,
+    }
+
+    for ax in [ax_test, ax_ref]:
+        ax.tick_params(axis=which, **target)
+
+    ax_test.clear()
+    ax_test.tick_params(axis=which, **target)
```

Comment:
I do not understand what this test is meant to catch. Unrolling the for loop, you do on `ax_test`
```
ax_test.tick_params(axis=which, **target)
ax_test.clear()
ax_test.tick_params(axis=which, **target)
```
which, irrespective of the `clear()` behavior, should be the same as only
```
ax_test.tick_params(axis=which, **target)
```
This is compared to 
```
ax_ref.tick_params(axis=which, **target)
```
Unless I'm mistaken somewhere above, this test is trivally true and doesn't actually test the clear behavior.

In https://github.com/matplotlib/matplotlib/issues/23806#issuecomment-1239069010 I had suggested a different pattern to catch the parameters that should survive `clear()`.